### PR TITLE
Sync noetic-devel -> ros2

### PR DESCRIPTION
**Pull request by henningkayser**

### CODE REVIEW

File: src/model.cpp [hunk 1]
```diff
@@ -563,10 +565,14 @@ void srdf::Model::loadDisabledCollisions(const urdf::ModelInterface& urdf_model,
   }
 }
 
-void srdf::Model::loadPassiveJoints(const urdf::ModelInterface& urdf_model, TiXmlElement* robot_xml)
+void srdf::Model::loadPassiveJoints(const urdf::ModelInterface& urdf_model, XMLElement* robot_xml)
 {
-  for (TiXmlElement* c_xml = robot_xml->FirstChildElement("passive_joint"); c_xml;
-       c_xml = c_xml->NextSiblingElement("passive_joint"))
+  for (XMLElement* c_xml = robot_xml->FirstChildElement("passive_joint"); c_xml; c_xml = c_xml->NextSiblingElement("pas"
```
Comment (henningkayser) [inline src/model.cpp]: This format looks weird, will fix it for noetic

Comment (henningkayser) [reply src/model.cpp]: #71

**Review by JafarAbdi — APPROVED** [review]

**Review by rhaschke — CHANGES_REQUESTED** [review]
The sync missed the latest ros2 commit and there are missing some changes from noetic-devel.
I'm looking into that right now.

**Comment by henningkayser** [comment]
> The sync missed the latest ros2 commit and there are missing some changes from noetic-devel.
> I'm looking into that right now.

@rhaschke I see, the merge was applied to PickNikRobotics:ros2 (no issue to fix that if you didn't already do so). What's missing from noetic-devel? It includes the latest `noetic-devel` head

**Comment by henningkayser** [comment]
ed8ce19 should be removed before merging

**Comment by rhaschke** [comment]
You essentially dropped https://github.com/ros-planning/srdfdom/pull/70/commits/4ccc89e8a783a035316ee5cdfb99418e7e407ce1 while merging. Looks like `ament` doesn't provide a mechanism to read the version from `package.xml`? At least in MoveIt2's main repo you hard-coded the version information:
https://github.com/ros-planning/moveit2/blob/6c80d778d82ad8a283be37bf4909008ece2c93b0/moveit_core/CMakeLists.txt#L2

**Comment by rhaschke** [comment]
Closing in favor of more complete #73.

**Pull request closed without merging**